GeoEntity.md

```diff
@@ -0,0 +1,280 @@
+#GeoEntity API
+
+&nbsp;
+
+#Contents
+
+- [Description](#description)
+- [All Countries Request](#all-countries-request)
+- [All Administrative Area Ones Request](#all-administrative-area-ones-request)
+- [All Administrative Area Twos Request](#all-administrative-area-twos-request)
+- [All Localities Request](#all-localities-request)
+- [All Postal Codes Request](#all-postal-codes-request)
+- [All Metropolitan Statistical Areas Request](#all-metropolitan-statistical-areas-request)
+- [Versioning](#versioning)
+
+&nbsp;
+----------------------------
+#Description
+
+This service provides aggregated geography entities. Currently the following requests could be made.
+- Request all countries in the world.
+- Request all administrative area ones in a specific country.
+- In a specific country ( and a specific first order of administrative area as optional), 
+	- Request all administrative area twos.
+	- Request all localities.
+	- Request all postal codes.
+	- Request all metropolitan statistical areas.
+	
+This service supports both HTTP/GET and POST methods.
+All responses with an HTTP status of 200 will consist of a JSON object with a top-level "data" node containing a node "entities", which is an list of entities as requested.
+
+&nbsp;
+----------------------------
+#All Countries Request
+- Request Structure:
+	- Target: http://localhost:8080/admin_area_1s (After adding to Routing layer, we update this)
+	- URL parameter: none
+	- Example URL:http://localhost:8080/countries
+	
+- Response Sample:
+```
+{
+  "data": {
+    "entities": [
+      {
+        "name": "Afghanistan",
+        "country_code": "AF",
+        "place_id": "ChIJbQL_-LZu0TgReNqWvg1GtfM",
+        "location": {
+          "lat": 33.93911,
+          "lng": 67.709953
+        },
+        "location_type": "APPROXIMATE",
+        "viewport": {
+          "northeast": {
+            "lat": 38.4908389,
+            "lng": 74.8898619
+          },
+          "southwest": {
+            "lat": 29.3772319,
+            "lng": 60.5170005
+          }
+        },
+        "formatted_address": "Afghanistan"
+     }
+  ]
+ }
+}
+```
+&nbsp;
+----------------------------
+#All Administrative Area Ones Request
+- Request Structure:
+	- Target: http://localhost:8080/admin_area_1s (After adding to Routing layer, we update this)
+	- URL parameter: 
+  
+    | Parameter  | Required | Description |
+    |----------------|----------------|-----------------|
+    | country        | Yes | A country name or two-letter ISO-3166 country code, not casing sensitive            | 
+    
+	- Example URL: http://localhost:8080/admin_area_1s?country=us
+	
+- Response Sample:
+```
+{
+  "data": {
+    "entities": [
+      {
+        "long_name": "Alaska",
+        "short_name": "AK",
+        "level": 1,
```

Comment:
But I guess it does not hurt. I just used one AdminArea class in several different places that they are serialzied/deserialized. It would look much more duplicate in code with the change.